src/engine/renderer/gl_shader.cpp

```diff
@@ -795,15 +798,21 @@ std::string     GLShaderManager::BuildGPUShaderText( Str::StringRef mainShaderNa
 			} );
 		if ( beginIt - line.begin() != int( position ) ) { // Signed/unsigned CI bullshit
 			shaderMain += line + "\n";
+			mainLine++;
 			continue;
 		}
 
+		// Inserted shader lines will start at 10000, 20000 etc. to easily tell them apart from the main shader code
+		shaderMain += "#line " + std::to_string( ( insertCount + 1 ) * 10000 ) + "\n";
+		insertCount++;
+
 		std::string shaderInsertPath = line.substr( position + 8, std::string::npos );
 		if ( shaderType == GL_VERTEX_SHADER ) {
 			shaderMain += GetShaderText( "glsl/" + shaderInsertPath + ".glsl" );
 		} else {
 			shaderMain += GetShaderText( "glsl/" + shaderInsertPath + ".glsl" );
 		}
+		shaderMain += "#line " + std::to_string( mainLine ) + "\n";
```

Comment:
It appears that mainLine is never incremented when there is a line with an insert directive. mainLine should be incremented on every iteration of the outer loop to keep the line numbers in sync with the source.

Reply:
This now works properly, eg:

```
// line 10
#line -1
#line 10000
// .. a bunch of code here
#line -1
// This line will have line number 11
```